.github/workflows/test_publish.yml

```
@@ -2,9 +2,8 @@
 name: Test PyPI Publish
 
 on:
-  push:
-    branches:
-      - main
+  release:
+    types: [published]
```

```suggestion
    types: [prereleased]
```

Ref: https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#release

---

done